var/spack/repos/builtin/packages/axom/package.py

```diff
@@ -0,0 +1,497 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+import os
+import socket
+from os.path import join as pjoin
+
+import llnl.util.tty as tty
+
+
+def cmake_cache_entry(name, value, comment=""):
+    """Generate a string for a cmake cache variable"""
+    return 'set({0} "{1}" CACHE PATH "{2}")\n\n'.format(name, value, comment)
+
+
+def cmake_cache_option(name, boolean_value, comment=""):
+    """Generate a string for a cmake configuration option"""
+
+    value = "ON" if boolean_value else "OFF"
+    return 'set({0} {1} CACHE BOOL "{2}")\n\n'.format(name, value, comment)
+
+
+def get_spec_path(spec, package_name, path_replacements={}, use_bin=False):
+    """Extracts the prefix path for the given spack package
+       path_replacements is a dictionary with string replacements for the path.
+    """
+
+    if not use_bin:
+        path = spec[package_name].prefix
+    else:
+        path = spec[package_name].prefix.bin
+
+    path = os.path.realpath(path)
+
+    for key in path_replacements:
+        path = path.replace(key, path_replacements[key])
+
+    return path
+
+
+class Axom(Package):
+    """Axom provides a robust, flexible software infrastructure for the development
+       of multi-physics applications and computational tools."""
+
+    maintainers = ['white238']
+
+    homepage = "https://github.com/LLNL/axom"
+    git      = "https://github.com/LLNL/axom.git"
+
+    version('master', branch='master', submodules=True, preferred=True)
```

Comment:
We usually like the newest stable release to be the default version, as master is a changing tip that isn't reproducible.

Reply:
Sounds good.  I removed the preferred all together. Am i right to assume that Spack will pick the highest numbered version if one isn't specified by the user?

Reply:
You are correct